applications/StructuralMechanicsApplication/custom_strategies/custom_builder_and_solver/explicit_builder_and_solver.hpp

```diff
@@ -124,61 +124,35 @@ class ExplicitBuilderAndSolver
         ElementsArrayType& r_elements       = rModelPart.Elements();
         ProcessInfo& r_current_process_info   = rModelPart.GetProcessInfo();
 
-        #ifdef _OPENMP
-            int number_of_threads = omp_get_max_threads();
-        #else
-            int number_of_threads = 1;
-        #endif
 
-        vector<unsigned int> node_partition;
-        OpenMPUtils::CreatePartition(number_of_threads, r_nodes.size(), node_partition);
-
-        vector<unsigned int> element_partition;
-        OpenMPUtils::CreatePartition(number_of_threads, r_elements.size(), element_partition);
-
-
-        #pragma omp parallel
+        typename NodesArrayType::iterator it_begin=r_nodes.ptr_begin();
+        #pragma omp parallel for 
+        for(size_t i=0;i<r_nodes.size();++i)
```

Comment:
minor: make the index to be "int" otherwise it will not compile in win.

also i would do

      auto it_begin=rModelPart.NodesBegin(); 

as this iterator may be slightly faster than the ptr_iterator

Reply:
but r_nodes.size() is a unsigned int -> I get warnings when compiling

Reply:
then cast it to an int:
`static_cast<int>(r_nodes.size())`

Reply:
👍